pkg/sdk/functions_def.go

```diff
@@ -0,0 +1,410 @@
+	"https://docs.snowflake.com/en/sql-reference/sql/create-function",
+	g.NewQueryStruct("CreateFunctionForScala").
+		Create().
+		OrReplace().
+		OptionalSQL("TEMPORARY").
+		OptionalSQL("SECURE").
+		SQL("FUNCTION").
+		IfNotExists().
+		Name().
+		ListQueryStructField(
+			"Arguments",
+			functionArgument,
+			g.ParameterOptions().Parentheses().NoEquals()).
+		OptionalSQL("COPY GRANTS").
+		OptionalQueryStructField(
+			"Returns",
+			functionReturns,
+			g.KeywordOptions().SQL("RETURNS"),
+		).
+		OptionalQueryStructField(
+			"NullOrNot",
+			functionNullOrNot,
+			g.KeywordOptions(),
+		).
+		SQL("LANGUAGE SCALA").
+		OptionalQueryStructField(
+			"StrictOrNot",
+			functionStrictOrNot,
+			g.KeywordOptions(),
+		).
+		OptionalQueryStructField(
+			"VolatileOrNot",
+			functionVolatileOrNot,
+			g.KeywordOptions(),
+		).
+		OptionalTextAssignment("RUNTIME_VERSION", g.ParameterOptions().SingleQuotes()).
+		OptionalTextAssignment("COMMENT", g.ParameterOptions().SingleQuotes()).
+		ListQueryStructField(
+			"Imports",
+			functionImports,
+			g.ParameterOptions().Parentheses().SQL("IMPORTS"),
+		).
+		ListQueryStructField(
+			"Packages",
+			functionPackages,
+			g.ParameterOptions().Parentheses().SQL("PACKAGES"),
+		).
+		OptionalTextAssignment("HANDLER", g.ParameterOptions().SingleQuotes()).
+		OptionalTextAssignment("TARGET_PATH", g.ParameterOptions().SingleQuotes()).
+		OptionalQueryStructField(
+			"FunctionDefinition",
+			functionDefinition,
+			g.ParameterOptions().NoEquals().SingleQuotes().SQL("AS"),
+		).
+		WithValidation(g.ValidIdentifier, "name"),
+).CustomOperation(
+	"CreateFunctionForSQL",
+	"https://docs.snowflake.com/en/sql-reference/sql/create-function",
+	g.NewQueryStruct("CreateFunctionForSQL").
+		Create().
+		OrReplace().
+		OptionalSQL("TEMPORARY").
+		OptionalSQL("SECURE").
+		SQL("FUNCTION").
+		IfNotExists().
+		Name().
+		ListQueryStructField(
+			"Arguments",
+			functionArgument,
+			g.ParameterOptions().Parentheses().NoEquals()).
+		OptionalSQL("COPY GRANTS").
+		OptionalQueryStructField(
+			"Returns",
+			functionReturns,
+			g.KeywordOptions().SQL("RETURNS"),
+		).
+		OptionalQueryStructField(
+			"NullOrNot",
+			functionNullOrNot,
+			g.KeywordOptions(),
+		).
+		OptionalQueryStructField(
+			"VolatileOrNot",
+			functionVolatileOrNot,
+			g.KeywordOptions(),
+		).
+		OptionalSQL("MEMOIZABLE").
+		OptionalTextAssignment("COMMENT", g.ParameterOptions().SingleQuotes()).
+		OptionalQueryStructField(
+			"FunctionDefinition",
+			functionDefinition,
+			g.ParameterOptions().NoEquals().SingleQuotes().SQL("AS"),
+		).
+		WithValidation(g.ValidIdentifier, "name"),
+).AlterOperation(
+	"https://docs.snowflake.com/en/sql-reference/sql/alter-function",
+	g.NewQueryStruct("AlterFunction").
+		Alter().
+		SQL("FUNCTION").
+		IfExists().
+		Name().
+		ListQueryStructField(
+			"ArgumentTypes",
+			functionArgumentType,
+			g.ParameterOptions().Parentheses().NoEquals()).
+		OptionalQueryStructField(
```

Comment:
I think we wanted to keep them as close as possible to the documentation (if they're on the separate lines then in code we should have operations for each of them instead of groupped up), but I would leave it as is, just have a validation that would prevent setting multiple set options.

Reply:
moved all actions inline